client/src/Components/Annotation/AnnotationList.tsx

```diff
@@ -54,7 +55,7 @@ export const Annotations = ({ meta, totalIQSamples, updateSpectrogram }) => {
 
       // Get the start and end dates
       const startDate = currentMetadata.captures[0]['core:datetime'];
-      const endDate = calculateDate(currentMetadata.captures[0]['core:datetime'], totalIQSamples, sampleRate);
+      const endDate = calculateDate(currentMetadata.captures[0]['core:datetime'], blob.totalIQSamples, sampleRate);
```

Comment:
Interresting to know that this is not being able to calculate form the metadata... maybe something that could be added to the spec.